clustered/integration-test/src/test/java/org/ehcache/clustered/replication/DuplicateTest.java

```diff
@@ -136,8 +136,7 @@ public void duplicateAfterFailoverAreReturningTheCorrectResponse() throws Except
             case "equals":
               return proxy == args[0];
             default:
-              fail("Unexpected method call: " + method.getName());
-              return null;
+              throw new AssertionError("Unexpected method call: " + method.getName());
```

Comment:
Any reason you didn't use `org.junit.Assert#fail(java.lang.String)`?

Reply:
I think I started with the move to Hamcrest over AssertJ (since the eventually method works with the former). In the process I would have just moved to my preferred pattern. I'm not a fan of unreachable statements.